[iOS] Extend DisneyPlus quirk to cover general iOS use cases

brentfulgham opened the pull request:
#### 16f7d1ba779bd2345f2a5ff9808f078d00886a9f
<pre>
[iOS] Extend DisneyPlus quirk to cover general iOS use cases
<a href="https://bugs.webkit.org/show_bug.cgi?id=291648">https://bugs.webkit.org/show_bug.cgi?id=291648</a>
&lt;<a href="https://rdar.apple.com/149344484">rdar://149344484</a>&gt;

Reviewed by Tim Nguyen.

Extend the cases where an existing DisneyPlus quirk is needed to cover more use cases.
This was originally thought be needed in a narrow case, but testing shows it&apos;s needed
in all iOS uses.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldHideCoarsePointerCharacteristics const):
(WebCore::Quirks::determineRelevantQuirks):
* Source/WebCore/page/QuirksData.h:

Canonical link: <a href="https://commits.webkit.org/293785@main">https://commits.webkit.org/293785@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3320334f1ea9fe7a99769a653d7247c32d755074

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99916 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19564 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9850 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/105079 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50497 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101957 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19869 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27999 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/76092 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/33144 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102923 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/15157 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90229 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56451 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14961 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/8214 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49866 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84887 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8303 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107404 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/27029 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19746 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/85010 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/27391 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86429 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84533 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21473 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29213 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6933 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/20848 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26966 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/32189 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26777 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30093 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/28336 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 reviewed (commented):
On Line 2112 in Quirks.cpp, you might have to update `#if ENABLE(DESKTOP_CONTENT_MODE_QUIRKS)` flag as well.

karlcow reviewed (commented):
Non reviewer review. LGTM!

brentfulgham committed:
[iOS] Extend DisneyPlus quirk to cover general iOS use cases
https://bugs.webkit.org/show_bug.cgi?id=291648
<rdar://149344484>

Reviewed by Tim Nguyen.

Extend the cases where an existing DisneyPlus quirk is needed to cover more use cases.
This was originally thought be needed in a narrow case, but testing shows it's needed
in all iOS uses.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldHideCoarsePointerCharacteristics const):
(WebCore::Quirks::determineRelevantQuirks):
* Source/WebCore/page/QuirksData.h:

Canonical link: https://commits.webkit.org/293785@main
